docs/variants/hardkernel_odroid_h4/initial-deployment.md

```diff
@@ -0,0 +1,38 @@
+# Initial deployment
+
+Initial flashing of Dasharo firmware can be done from Linux using flashrom with
+the internal programmer or using the Dasharo Tools Suite included in the Dasharo
+Pro Package subscription (formerly Dasharo Entry Subscription). An instruction
+on how to use the DTS can be found in the [DTS documentation](../../dasharo-tools-suite/documentation.md#dasharo-zero-touch-initial-deployment)
+
+This document describes the process of building, installing and running
+flashrom on Ubuntu 24.04.
+
+## Build flashrom
+
+Please follow generic guide for [Dasharo flashrom fork](../../osf-trivia-list/deployment.md#how-to-install-dasharo-flashrom-fork).
+
+## Reading flash contents
+
+Always prepare a backup of the current firmware image. To read from the flash
+and save it to a file (`dump.rom`), execute the following command:
+
+```bash
+flashrom -p internal -r dump.rom
+```
+
+If you forgot to do this, you can get the original, proprietary
+firmware from the [ODROID wiki website](https://wiki.odroid.com/odroid-h4/hardware/h4_bios_update#bios_release)
+ to restore it in case it is bricked. Using the firmware other than
+your own backup should be the last resort.
+
+## Flashing Dasharo
+
+To flash Dasharo on the platform, execute the following command - replace `[path]`
+with the path to the Dasharo image you want to flash, e.g. `hardkernel_odroid_h4_v0.9.0.rom`.
+
+```bash
+sudo flashrom -p internal -w [path] --ifd -i bios
```

Comment:
Are we sure particle flashing is what we should do for initial deployment? I have doubts because using partial flashing from AMI to Dasharo gave me no support for the network. Then partial flashing from Dasharo to Dasharo leads to serious bricking. The reset of CMOS + G3 helped, but I was pretty concerned when switching SPI to AMI, which didn't boot (it looks like a lack of CMOS reset). Let's make sure this section is robust.

Reply:
> I have doubts because using partial flashing from AMI to Dasharo gave me no support for the network.

Any more details on it? iPXE has to be explicitly enabled in the setup. 

> Then partial flashing from Dasharo to Dasharo leads to serious bricking.

Happened to me as well, but also after AMI->Dasharo transition. This PR should fix the bricking: https://github.com/Dasharo/coreboot/pull/569

Reply:
> Any more details on it? iPXE has to be explicitly enabled in the setup.

After booting to DTS v2.0.0, I had no network interface. I cannot reproduce that now, but I should analyze it more deeply when testing over the next 1-2 days.

Reply:
It happened again while doing Dasharo->AMI transition:

```sh
bash-5.2# flashrom -p internal -w logs/rom.bin
flashrom v1.2-1037-g5b4a5b40 on Linux 6.6.21-yocto-standard (x86_64)
flashrom is free software, get the source code at https://flashrom.org

Using clock_gettime for delay loops (clk_id: 1, resolution: 1ns).
coreboot table found at 0x76933000.
Found chipset "Intel Alder Lake-N".
Enabling flash write... Warning: Setting BIOS Control at 0xdc from 0x8b to 0x89 failed.
New value is 0x8b.
SPI Configuration is locked down.
FREG0: Flash Descriptor region (0x00000000-0x00000fff) is read-write.
FREG1: BIOS region (0x00600000-0x00ffffff) is read-write.
FREG2: Management Engine region (0x00001000-0x00413fff) is read-write.
PR0: Warning: 0x00b00000-0x00ffffff is read-only.
At least some flash regions are write protected. For write operations,
you should use a flash layout and include only writable regions. See
manpage for more details.
OK.
Found Programmer flash chip "Opaque flash chip" (16384 kB, Programmer-specific) on internal.
Reading old flash chip contents... done.
Erasing and writing flash chip... Transaction error between offset 0x00b00000 and 0x00afffff (= 0x00b00000 + -1)!
Reading current flash chip contents... done. Looking for another erase function.
Looking for another erase function.
Looking for another erase function.
Looking for another erase function.
Looking for another erase function.
Looking for another erase function.
Looking for another erase function.
No usable erase functions left.
FAILED!
Uh oh. Erase/write failed. Checking if anything has changed.
Reading current flash chip contents... done.
Apparently at least some data has changed.
Your flash chip is in an unknown state.
Get help on IRC (see https://www.flashrom.org/Contact) or mail
flashrom@flashrom.org with the subject "FAILED: <your board name>"!-------------------------------------------------------------------------------
DO NOT REBOOT OR POWEROFF!
```

This is my fault since I didn't remove all necessary locks before flashing, but the critical question is how to deal with such a situation correctly. PR0 range is weird, why it does not cover whole bios region?

Reply:
I guess one way to deal with the problem is to write Dasharo again, go to the setup menu, disable protection, and then do the recovery.

Reply:
During training, @mkopec also faced a similar issue with the lack of an Ethernet interface, it seem to be some problem with power cycles because after G3, things started to work correctly.

Reply:
We might say to perform a G3 an/or cmos reset if necessary. I am not sure what other changes I can suggest.

Reply:
> This is my fault since I didn't remove all necessary locks before flashing, but the critical question is how to deal with such a situation correctly. 

Platform should still be bootable

> PR0 range is weird, why it does not cover whole bios region?

It covers the WP_RO region, Vboot's recovery. So theoretically it should boot from recovery (provided Dasharo works with AMI's ME blobs)